src/models/device.ts

```diff
@@ -3135,22 +3119,30 @@ const getDeviceModel = function (
 			 * balena.models.device.history.getAllByApplication(999999, { fromDate: subDays(new Date(), 20), toDate: subDays(new Date(), 10)})
 			 *
 			 */
-			async getAllByApplication(
+			async getAllByApplication<
+				T extends ODataOptionsWithoutCount<DeviceHistory['Read']>,
+			>(
 				slugOrUuidOrId: string | number,
 				{
 					fromDate = subDays(new Date(), DEFAULT_DAYS_OF_REQUESTED_HISTORY),
 					toDate,
 					...options
-				}: PineOptions<DeviceHistory> & { fromDate?: Date; toDate?: Date } = {},
-			): Promise<DeviceHistory[]> {
+				}: T & {
```

Comment:
It seems like it would make sense to separate the pinejs-client options from the balena-sdk options into two separate parameters, it would make the typing/code cleaner and when using it I think it would also make it clearer that `fromDate`/`toDate` are not things that exist in the model or are expected to potentially pass through to pinejs-client

(and elsewhere that follows the pattern)

Reply:
done (follow up commit I will pull up when finishing rebasing)